packages/test/test-end-to-end-tests/src/test/gc/gcContainerRuntimeCompat.spec.ts

```diff
@@ -124,29 +62,25 @@ describeFullCompat.skip("GC summary compatibility tests", (getTestObjectProvider
             [innerRequestHandler],
             runtimeOptions,
         );
+        return provider.createContainer(runtimeFactory, { configProvider: mockConfigProvider() });
     }
 
     beforeEach(async () => {
-        provider = getTestObjectProvider();
-        // Wrap the document service factory in the driver so that the `uploadSummaryCb` function is called
-        // every time the summarizer client uploads a summary.
-        (provider as any)._documentServiceFactory = wrapDocumentServiceFactory(
-            provider.documentServiceFactory,
-            uploadSummaryCb,
-        );
-
-        mainContainer = await createContainer(runtimeFactory);
-        dataStoreA = await requestFluidObject<TestDataObject>(mainContainer, "default");
-
-        await provider.ensureSynchronized();
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        mainContainer = await createContainer();
+        dataStoreA = await requestFluidObject<ITestFluidObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
     });
 
-    afterEach(() => {
-        latestAckedSummary = undefined;
-        latestSummaryContext = undefined;
-        latestUploadedSummary = undefined;
-        provider.reset();
-    });
+    async function createSummarizer(version: number, summaryVersion?: string) {
```

Comment:
I would like to be able to pass in some form of summaryOptions like canceling summaryHeuristics